suse_migration_services/units/product_setup.py

```diff
@@ -0,0 +1,56 @@
+# Copyright (c) 2018 SUSE Linux LLC.  All rights reserved.
+#
+# This file is part of suse-migration-services.
+#
+# suse-migration-services is free software: you can redistribute it and/or
+# modify it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# suse-migration-services is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with suse-migration-services. If not, see <http://www.gnu.org/licenses/>
+#
+import os
+
+# project
+from suse_migration_services.command import Command
+from suse_migration_services.defaults import Defaults
+from suse_migration_services.logger import log
+
+from suse_migration_services.exceptions import (
+    DistMigrationProductSetupException
+)
+
+
+def main():
+    """
+    DistMigration setup product
+
+    Synchronize bind mounted etc/products.d data into the migrated system
+    """
+    root_path = Defaults.get_system_root_path()
+
+    try:
+        products_metadata = os.sep.join(
+            [root_path, 'etc', 'products.d']
+        )
+        log.info('Umounting {0}'.format(products_metadata))
+        Command.run(
+            ['umount', products_metadata]
+        )
+        log.info('Syncing product data to migrated system')
+        Command.run(
+            [
+                'rsync', '-zav', '--delete', '/etc/products.d/',
+                products_metadata + os.sep
+            ]
+        )
```

Comment:
Please add a comment that this may only work as long as FATE#320882 is not implemented. Once we support multiple registration servers this may break depending on the implementation of migration.

Yes it makes no sense to try and upgrade multiple sources from different servers at the same time, but if that happens then this implementation would clobber important files for the next migration.

Consider the following use case once FATE#320882 is implemented:
An on demand instance is also registered against SCC for life patching

Then there would be a life-patching.prod file in the products.d directory. If the migration runs in phases per update server and this code runs first then the life-patching.prod file wold be removed and that upgrade against SCC would fail.

There are way around this, and I can help with the implementation if we want to do that now. If we want to wait to handle that use case later then we should at least leave ourselves a note about this problem.

Reply:
Thanks for the talk on mumble. I've added comment blocks and hope they make the point clear